### ISSUE TITLE
[Suggestion] Loot Table

### ISSUE THREAD

**Issue by Pro457** [SimonFlash/TeslaCrate#5]
Model: https://github.com/Codisimus/PhatLoots

To put this simply, PhatLoots functioned like a box full of items. Every single item in the box had a probability assigned to it based off of a 100% scale percentage. When a person would open the box using a key, EVERY SINGLE ITEM inside the box was "rolled" for using the assigned percentage, and if it rolled successfully the item was given to the user. 

If you do not understand what I mean, I can show you an example. Essentially you would receive multiple items/commands/etc. all from ONE key. It was completely random every single time what you would receive, but EVERYTHING in the loot table was ALWAYS rolled for.

**Comment by SimonFlash** [comment]
Long story short, I need to do a better job of keeping up with issues - I've been meaning to do this for a while now, but a clean solution has evaded me until now (probably because I tried coding it xD).

This specific feature could be done with a simple `roll-all` setting in the crate. This could be a simple `true`/`false`, or it could the weight that we're comparing this to - for example, `roll-all=100` would mean that a weight of `30` means a 30% chance, while `roll-all=300` would be a 10% chance. I'm not sure if this additional complexity is worth it (figure most would have it be percentages), but it does specify that weight is not a direct indicator of percentage alone.

The next piece of this is that I'd also like a system where a crate can process it's rewards multiple times. This could be done by having a `roll-quantity` variable, but looking ahead it might be better to have a `roll-min` and `roll-max` setting. I would need to look at how I format messages set out as this could easily get spammy.

With all that being said, here's what I'd be looking at:

```java
roll-all=100 //percentages
roll-min=1 //minimum 1 reward (0 is allowed)
roll-max=3 //maximum 3 rewards (even distribution between 1, 2, and 3)
```

I apologize for having this be extended for so long, but hopefully future users can make use of this as well if you've moved on.

**Comment by SimonFlash** [comment]
Implemented as of v2.1.1. Instead of having `roll-all=100` or something, this instead uses `weight-sum=100` and `roll-all=true`. There's still some room for improvement in my opinion, but it's functional!

With regards to the full loot table aspect of things, I still believe that there's definitely a use for this type of functionality in Sponge and would be best suited as it's own plugin.
